### PR TITLE
qa(q8): reject output AMX prefill lane

### PR DESCRIPTION
**Pull request by timtoole02**
## Summary
- add Linux x86_64 AMX parity coverage for the existing default-off `CAMELID_X86_Q8_OUTPUT_AMX_PREFILL` output-prefill helper
- clear `CAMELID_X86_Q8_AMX_REPACK` in the dense diagnostic env helper so the new focused parity test fails closed
- record a scrubbed Ubuntu x86_64 same-host rejection note and link it from the Ubuntu x86 Q8 performance doc

## Validation
- Ubuntu Linux x86_64: `cargo test --locked --lib x86_q8_output_amx_prefill_matches_rows4_matmul_when_supported -- --nocapture`
- Ubuntu Linux x86_64: `cargo test --locked --lib x86_q8_output_packed_rows4_matmul_is_plan_gated_and_prefill_limited -- --nocapture`
- Ubuntu Linux x86_64: release builds on current `main` and candidate head
- Ubuntu Linux x86_64: one-token parity for baseline and candidate
- Ubuntu Linux x86_64: same-host benchmark with `warmup=1`, `repeats=3`, `max_tokens=16`, unique prompt, and marker guard enabled
- `cargo fmt --all --check`

## Same-host evidence
- Baseline packed-rows4 lane with `CAMELID_X86_Q8_OUTPUT_PACKED_ROWS4_MATMUL=on`: Camelid TTFT `3283.36 ms`, total `3283.64 ms`, backend first content `2798.67 ms`; llama.cpp TTFT `377.79 ms`, total `570.25 ms`
- Candidate lane with `CAMELID_X86_Q8_AMX_REPACK=on CAMELID_X86_Q8_OUTPUT_AMX_PREFILL=on`: Camelid TTFT `4024.04 ms`, total `4024.35 ms`, backend first content `3523.67 ms`; llama.cpp TTFT `377.82 ms`, total `570.55 ms`
- One-token parity stayed exact (`C` vs `C`) for both baseline and candidate; candidate one-token Camelid TTFT still worsened from `5788.13 ms` to `7808.46 ms`
- Marker guard passed for every measured 16-token run
- Current telemetry still records the logits work under the shared `q8_0_borrowed_packed_rows4` bucket, so this PR treats the AMX-prefill helper as rejected until a distinct route label and a same-host wall-clock win both exist

## Decision
- Keep `CAMELID_X86_Q8_OUTPUT_AMX_PREFILL` default-off
- Reject the current output AMX prefill slice for retention/default-on/support-claim purposes on the current Ubuntu x86_64 validator

### CODE REVIEW

**Comment by timtoole02** [comment]
Fresh same-host Ubuntu x86_64 rerun on the current validator, using a clean disposable checkout and shared target storage.

Artifact label: `camelid-pr130-rerun-20260524T182621Z`

Validation:
- `cargo fmt --all --check`
- `cargo test --locked --lib x86_q8_output_amx_prefill_matches_rows4_matmul_when_supported -- --nocapture`
- `cargo test --locked --lib x86_q8_output_packed_rows4_matmul_is_plan_gated_and_prefill_limited -- --nocapture`
- release builds for `origin/main` (`07c912cf3041`) and this PR head (`8436a7473754`)
- one-token same-host check (`max_tokens=1`, unique prompt)
- same-host benchmark (`warmup=1`, `repeats=3`, `max_tokens=16`, unique prompt, marker guard on)

Results:
- One-token exact text still matched on base and candidate: `"C"` vs `"C"`
- One-token Camelid TTFT worsened from `6503.45 ms` on base to `8486.24 ms` on the candidate
- 16-token base: Camelid `TTFT 3162.06 ms`, total `3162.38 ms`, backend first content `2705.00 ms`; llama.cpp `TTFT 599.19 ms`, total `952.20 ms`
- 16-token candidate: Camelid `TTFT 3874.20 ms`, total `3874.43 ms`, backend first content `3387.67 ms`; llama.cpp `TTFT 598.71 ms`, total `953.76 ms`
- Candidate delta vs base: `+712.14 ms` TTFT (`+22.52%`)
- Marker guard passed on both measured runs

This rerun reproduces the rejection cleanly: parity is exact, but the AMX-prefill path is materially slower than the retained packed-rows4 baseline on the active Ubuntu x86 Q8 lane, so it should remain default-off and non-retained.
